pumpkin/src/client/player_packet.rs

```diff
@@ -400,6 +400,10 @@ impl Client {
     }
 
     pub fn handle_set_creative_slot(&mut self, _server: &mut Server, packet: SSetCreativeSlot) {
+        let gamemode = self.player.as_ref().unwrap().gamemode;
+        if gamemode != GameMode::Creative {
+            self.kick("CHEATER")
+        }
         let inventory = &mut self.player.as_mut().unwrap().inventory;
```

Comment:
Why we should `&mut self.player.as_mut()` twice when we also could just do it once ?

Reply:
because the borrow checker yells at me if i try to have a mut ref to player (that gets used later, so it can't be dropped) when doing self.kick (which requires &mut)

Reply:
> because the borrow checker yells at me if i try to have a mut ref to player (that gets used later, so it can't be dropped) when doing self.kick (which requires &mut)

Use
```rust
pub fn handle_set_creative_slot(&mut self, _server: &mut Server, packet: SSetCreativeSlot) {
        let player = self.player.as_mut().unwrap();

        if player.gamemode != GameMode::Creative {
            self.kick("Invalid action, you can only do that if you are in creative");
            return;
        }

        player
            .inventory
            .set_slot(packet.slot as usize, packet.clicked_item.to_item(), false);
    }
 ```